src/components/ModalWithConfirmation.js

```diff
@@ -0,0 +1,30 @@
+import Modal from "./Modal";
+
+class ModalWithConfirmation extends Modal {
+  constructor(modalSelector, handleSubmit) {
+    super({ modalSelector });
+    this._handleSubmit = handleSubmit;
+    this._submitButton = this._modalElement.querySelector(".modal__submit");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._submitButton.addEventListener("click", (event) => {
+      event.preventDefault();
+      this._handleSubmit(this._cardElement, this._cardId);
+      this.close();
+    });
+  }
+
+  open(cardElement, cardId) {
+    this._cardElement = cardElement;
+    this._cardId = cardId;
+    super.open();
+  }
+
+  close() {
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.